### ISSUE TITLE
splitEdgeTriangular causes inputVertexPositions to double in size

### ISSUE THREAD

**Issue by icethrush** [nmwsharp/geometry-central#57]
It looks like `ManifoldSurfaceMesh::splitEdgeTriangular` causes something strange to happen to the `inputVertexPositions` buffer. Here's the simplest example on which I see it:
```c++
void myCallback()
{
  if (ImGui::Button("Split one edge"))
  {
    std::cout << "Before: " << geometry->inputVertexPositions.size() << " positions in inputVertexPositions" << std::endl;
    std::cout << "Before: " << mesh->nVertices() << " vertices according to nVertices()" << std::endl;

    // Split an edge and set the new vertex to be at the midpoint
    Edge e = mesh->edge(1);
    Vector3 p1 = geometry->inputVertexPositions[e.halfedge().vertex()];
    Vector3 p2 = geometry->inputVertexPositions[e.halfedge().twin().vertex()];
    Vector3 newPos = (p1 + p2) / 2;
    Halfedge he = mesh->splitEdgeTriangular(e);
    Vertex newV = he.vertex();
    geometry->inputVertexPositions[newV] = newPos;

    std::cout << "After: " << geometry->inputVertexPositions.size() << " positions in inputVertexPositions" << std::endl;
    std::cout << "After: " << mesh->nVertices() << " vertices according to nVertices()" << std::endl;

    geometry->refreshQuantities();

    // Register the mesh with polyscope
    psMesh = polyscope::registerSurfaceMesh(
        "meshWithSplit",
        geometry->inputVertexPositions, mesh->getFaceVertexList(),
        polyscopePermutations(*mesh));
  }
}
```
I replaced the sample `myCallback()` function in the current version of the template project (https://github.com/nmwsharp/gc-polyscope-project-template) with this one. When I run the project on any mesh and click the resulting "Split one edge" button, the following console output results:
![Screenshot from 2020-12-16 12-04-13](https://user-images.githubusercontent.com/2643021/102382330-5646ee00-3f98-11eb-82a0-2b90e8a44805.png)
The `inputVertexPositions` buffer has now doubled in size (not capacity, but actual reported size), but `nVertices()` indicates that only one vertex was actually added. The resulting mismatch produces an error when attempting to reregister the mesh in polyscope.

It looks like the polyscope error only started being raised after [this commit](https://github.com/nmwsharp/geometry-central/commit/99e9dfaea80dac3ff38803ffa633512d7e3a083d), but the underlying behavior has been silently occurring before then. Prior to that commit, this was what would happen:
![Screenshot from 2020-12-16 12-44-31](https://user-images.githubusercontent.com/2643021/102386574-36fe8f80-3f9d-11eb-9c4f-a8ccbae4f29d.png)
![Screenshot from 2020-12-16 12-44-26](https://user-images.githubusercontent.com/2643021/102386112-ad4ec200-3f9c-11eb-8920-a832446d35ba.png)
Polyscope would think that there were twice as many vertices, but it would allow the registration to go through without an error. So the issue would only be noticed if one was paying attention to the vertex count on the UI (which apparently I hadn't been).

**Comment by icethrush** [comment]
[This commit](https://github.com/nmwsharp/geometry-central/commit/a6e1e25a4f1c4fe63f34b88751ecb08a0b0769b8) appears to be the most recent one where vertex counts are consistent across all data structures in both libraries.
![Screenshot from 2020-12-16 13-32-01](https://user-images.githubusercontent.com/2643021/102391047-323cda00-3fa3-11eb-946f-555a5957941f.png)
![Screenshot from 2020-12-16 13-27-54](https://user-images.githubusercontent.com/2643021/102391062-3668f780-3fa3-11eb-8dce-78bbb98ae370.png)
The commit immediately after this one causes polyscope to start thinking that mesh vertex count has doubled, and the commit flagged in the OP is when polyscope starts rejecting the mesh outright. (All of these scenarios are built against the same, most recent commit of polyscope.)

**Comment by nmwsharp** [comment]
Thanks for the report!

Should be fixed with 7b6862d900771c9ca90e46e67a7648c803d1df8b. 

An explanation for anyone who comes across this:

For the container size; this is the expected behavior. Containers resize lazily as the mesh is mutated to amortize allocation cost.

For Polyscope, one should always call `compress()` before sending a mesh to Polyscope, to ensure buffers are dense. However, that wasn't working here because of the semantics of compression w/ insertions, which are fixed in the commit above.
